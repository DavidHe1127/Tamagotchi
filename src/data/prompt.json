{
  "title": "What can I do for you?",
  "options": ["Feed me with snags", "Put me to sleep", "How are you?"]
}
