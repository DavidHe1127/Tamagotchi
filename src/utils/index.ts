const buildPrompt = (title: string, options: string[]): string => {
  return options.reduce((prev, next, index) => {
    prev += `${index + 1}. ${next}\n`;
    return prev;
  }, `${title}\n\n`);
};

const msg = (msgText: string): void =>
  console.log('\x1b[32m%s\x1b[0m', msgText);

export default {
  buildPrompt,
  msg,
};
