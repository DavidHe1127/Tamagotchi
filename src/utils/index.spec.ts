import utils from './index';
import * as promptData from '../data/prompt.json';

describe('test utils', () => {
  it('Build prompt with given data', () => {
    const prompt = utils.buildPrompt(promptData.title, promptData.options);
    expect(prompt).toMatchSnapshot();
  });

  it('print message in expected format', () => {
    const format = '\x1b[32m%s\x1b[0m';
    const msg = 'this is a test';

    jest.spyOn(global.console, 'log');
    utils.msg(msg);

    expect(console.log).toHaveBeenCalledWith(format, msg);
  });
});
