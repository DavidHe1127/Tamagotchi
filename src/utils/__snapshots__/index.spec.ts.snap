// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test utils Build prompt with given data 1`] = `
"What can I do for you?

1. Feed me with snags
2. Put me to sleep
3. How are you?
"
`;
